### PR TITLE
Bug/CCP-228

### PR DESCRIPTION
**Pull request by mkarthik90**

### CODE REVIEW

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/IConceptWrapperAddService.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package edu.asu.conceptpower.app.service;
+
+import java.util.Map;
+
+public interface IConceptWrapperAddService {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/IConceptWrapperAddService.java]: needs javadoc

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperAddService.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.stereotype.Service;
+
+import edu.asu.conceptpower.app.core.IConceptListManager;
+import edu.asu.conceptpower.app.core.IConceptTypeManger;
+import edu.asu.conceptpower.app.service.IConceptWrapperAddService;
+import edu.asu.conceptpower.core.ConceptList;
+import edu.asu.conceptpower.core.ConceptType;
+
+@Service
+public class ConceptWrapperAddService implements IConceptWrapperAddService {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperAddService.java]: why is that an "addService"? doesn't add anything.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java [hunk 3]
```diff
@@ -112,41 +105,40 @@ public String prepareConceptWrapperAdd(HttpServletRequest req, ModelMap model) {
...
      */
     @RequestMapping(value = "auth/conceptlist/addconceptwrapper/add", method = RequestMethod.POST)
-    public String addConcept(HttpServletRequest req, Principal principal, Model model)
-            throws DictionaryDoesNotExistException, DictionaryModifyException, LuceneException, IllegalAccessException, IndexerRunningException {
-
-        if (req.getParameter("word") == null || req.getParameter("word").trim().isEmpty()) {
-            req.setAttribute("errormsg", "You have to enter a word.");
-            return "forward:/auth/conceptlist/addconceptwrapper";
+    public String addConcept(@Validated @ModelAttribute ConceptWrapperAddBean conceptWrapperAddBean,
+            BindingResult result, HttpServletRequest request, Principal principal, Model model)
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: is the request parameter used anywhere?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java [hunk 4]
```diff
@@ -161,7 +153,8 @@ public String addConcept(HttpServletRequest req, Principal principal, Model mode
      * @throws IllegalAccessException
      */
     @RequestMapping(value = "/auth/conceptlist/addconceptwrapper/conceptsearch", method = RequestMethod.POST)
-    public String search(HttpServletRequest req, ModelMap model) throws LuceneException, IllegalAccessException {
+    public String search(@ModelAttribute ConceptWrapperAddBean conceptWrapperAddBean, HttpServletRequest req,
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: is the conceptWrapperAddBean used anywhere?

File: Conceptpower+Spring/src/main/webapp/WEB-INF/views/auth/concepts/ConceptWrapperAddView.jsp [hunk 5]
```diff
@@ -291,6 +290,51 @@
         $("#synonymsids").val(synonyms);
         $('#addedSynonnym').val(synonyms);
     };
+
+    <!-- This has been added to populate the selected wordnet ids-->
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/webapp/WEB-INF/views/auth/concepts/ConceptWrapperAddView.jsp]: I"m pretty sure this is not how you write javascript comments

**Comment by jdamerow** [comment]
also needs test cases for the new classes

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperAddService.java (hunk 2 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperAddService.java]: Changed to ConceptWrapperService

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java (hunk 4 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: Added conceptWrapperAddBean to model object to be used as a commandObject in the jsp form.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperService.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        return types;
+    }
+
+    /**
+     * This method fetches all the concept lists within the Conceptpower system
+     * and returns a map containing the concept list name as the key and concept
+     * list name as the value as well.
+     * 
+     * If there are no ConceptLists in the Conceptpower system, then an empty
+     * map is returned.
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperService.java]: this kind of information should be in interface javadoc

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java (hunk 4 above)
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptWrapperAddController.java]: Ideally this method would be an ajax call and not reload the whole form, but for now let's leave it that way.

File: Conceptpower+Spring/src/main/webapp/WEB-INF/views/auth/concepts/ConceptWrapperAddView.jsp [hunk 7]
```diff
@@ -291,6 +290,51 @@
...
+                        
+                        html += '<a id=' +conceptEntry.id + ' data-toggle=\"modal\" data-target=\"#detailsModal\"';
+                        html += 'data-conceptid=' + conceptEntry.id +'>'; 
+                        html += '<h5>' + conceptEntry.lemma + '</h5>';
+                        html += "</a>";
+                        
+                        html += '<h5> [' + conceptEntry.id + ']' + '</h5>';
+                        html += '<p>' + conceptEntry.description + '</p></div>';     
+                    });             
+                    console.log(html);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/webapp/WEB-INF/views/auth/concepts/ConceptWrapperAddView.jsp]: remove logger

File: Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    @InjectMocks
+    private ConceptWrapperService conceptWrapperService;
+
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void test_fetchAllConceptTypes_successForConceptLists() {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperServiceTest.java]: why is this test called test_fetchAllConceptTypes_... if it's not testing that method?

File: Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+        Assert.assertEquals(2, conceptListMap.size());
+        Assert.assertEquals(firstConceptList.getConceptListName(),
+                conceptListMap.get(firstConceptList.getConceptListName()));
+        Assert.assertEquals(secondConceptList.getConceptListName(),
+                conceptListMap.get(secondConceptList.getConceptListName()));
+
+    }
+
+    @Test
+    public void test_fetchAllConceptTypes_successForConceptListWithASingleEntry() {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperServiceTest.java]: same here

File: Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+        Map<String, String> conceptListMap = conceptWrapperService.fetchAllConceptLists();
+        Mockito.verify(conceptListManager).getAllConceptLists();
+        Assert.assertEquals(1, conceptListMap.size());
+        Assert.assertEquals(firstConceptList.getConceptListName(),
+                conceptListMap.get(firstConceptList.getConceptListName()));
+
+    }
+
+    @Test
+    public void test_fetchAllConceptTypes_successForEmptyConceptList() {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/test/java/edu/asu/conceptpower/app/service/impl/ConceptWrapperServiceTest.java]: same here

**Comment by mkarthik90** [comment]
@jdamerow  Addressed review comments.

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
run integration tests please

**Pull request merged**